addons/mail/models/mail_activity.py

```diff
@@ -251,9 +252,23 @@ def action_feedback(self, feedback=False):
             )
             message |= record.message_ids[0]
 
-            # Removing attachments res_id to prevent unlinking it
-            # We still want them in the message
-            self.env['ir.attachment'].search([('res_model', '=', self._name), ('res_id', 'in', self.ids)]).write({'res_id': False})
+        # Moving the attachments in the message
+        # TODO: Fix void res_id on attachment when you create
+        # an activity with an image directly see (/web_editor/attachment/add)
+        message.attachment_ids = self.env['ir.attachment'].search([
```

Comment:
`message` can be a recordset, so that won't work.

Do a `search_read` before looping on `self` (so you have a list matching the attachments and the activities), then add the attachment on `record.message_ids[0]`